### PR TITLE
CLIMATE-912 Upgrade mccSearch code from Python2 > 3

### PR DESCRIPTION
**Pull request by lewismc**
This issue addresses [CLIMATE-912](https://issues.apache.org/jira/browse/CLIMATE-912)

### CODE REVIEW

**Comment by lewismc** [comment]
@kwhitehall if you are able to pull this PR locally and try it out with some test data (and reference the data here so I can also try it out) it would be very much appreciated.

**Comment by kwhitehall** [comment]
Ack'ed @lewismc

**Comment by lewismc** [comment]
Thank you @kwhitehall

**Comment by lewismc** [comment]
Hi @kwhitehall I just went to look at some more OCW code and my local branch was set to this issue. I had to come here to remind myself what the issue was about. Did you get any time at all to take a look through this?

**Comment by kwhitehall** [comment]
Hey @lewismc 
Tested and found the following:

1. need to convert this spaces to tabs in this folder
2. updated code to reduce the number of datasets needed to test.
Code for test can be obtained at the this [Google Drive loc](https://drive.google.com/drive/folders/0B0kFBnuqk_GTczFiU3kwWDBXaFk?usp=sharing)

Note that I did a hacky update cause another error flagged in [dataset_processor.py](https://github.com/apache/climate/blob/master/ocw/dataset_processor.py#L1004-1005) by changing them to:
    spatial_indices_true = np.where(spatial_values.mask.all)
    spatial_indices_false = np.where(not spatial_values.mask.all)
I haven't included these as something to include in this PR as more investigation needs to be done to ensure that this doesn't break the code otherwise. 

The PR has been pushed to my local [here](https://github.com/kwhitehall/climate/pull/7). 
I'll work on pushing it to your PR. Do you have direction on this?

**Comment by lewismc** [comment]
Hi @kwhitehall I am sorry I didn't get back to you on this.
We need to update the PR to resolve conflicts. I will take a wee look on Monday.

File: mccsearch/code/mainProg.py [hunk 1]
```diff
@@ -55,35 +55,35 @@ def main():
     # -------------------------------------------------------------------------------------------------
 
     # let's go!
-    print "\n -------------- Read MERG Data ----------"
+    print("\n -------------- Read MERG Data ----------")
     mergImgs, timeList = mccSearch.readMergData(CEoriDirName)
-    print ("-" * 80)
+    print(("-" * 80))
```
Comment (cclauss) [inline mccsearch/code/mainProg.py]: Recommendation to change `print(("-" * 80))` to `print("-" * 80)` in eight places.  The double nested parens `(( ))` are not needed and only slow down your reader.  Repeated on lines: 60, 71, 73, 76, 78, 81, 84, 107.

File: mccsearch/code/mainProgTemplate.py [hunk 2]
```diff
@@ -59,36 +59,36 @@ def main():
     # -------------------------------------------------------------------------------------------------
 
     # let's go!
-    print "\n -------------- Read MERG Data ----------"
+    print("\n -------------- Read MERG Data ----------")
     mergImgs, timeList = mccSearch.readMergData(CEoriDirName)
-    print ("-" * 80)
+    print(("-" * 80))
```
Comment (cclauss) [inline mccsearch/code/mainProgTemplate.py]: Recommendation to change `print(("-" * 80))` to `print("-" * 80)` in eight places.  The double nested parens `(( ))` are not needed and only slow down your reader.  Repeated on lines: 64, 75, 77, 80, 82, 86, 89, 112.

**Pull request merged**